### ISSUE TITLE
Blurry Dynamic Objects in Painter Sequence (Techni Full Config) and Custom Pose Rendering

### ISSUE THREAD

**Issue by JunhyeongDoyle** [oppo-us-research/SpacetimeGaussians#84]
Hello, thank you for your amazing work on this project! I’ve been making good progress using the Techni full config for training on the Painter sequence. While the results are impressive for static backgrounds, I’m encountering an issue where dynamic objects (e.g., moving people) appear too blurry compared to the rest of the scene. I’ve experimented with the following, but I haven’t seen significant improvements:

Adjusting the training iterations (increased up to 200%).
Tweaking various config parameters
Applying different augmentation techniques, but the blurriness issue persists.

Question 1 :
Could you please suggest any tips or potential adjustments I could try to resolve this issue? Are there specific config parameters or training strategies you recommend to handle dynamic objects more effectively?

![00049](https://github.com/user-attachments/assets/cc62f84a-5379-446e-bda3-656a1fba1db8)

And I have one more question about custom viewport rendering with camera extrinsic parameters

I’d like to render a scene from a custom viewport by manually specifying the extrinsic parameters (X, Y, Z, Yaw, Pitch, Roll). I’m looking for guidance on which parts of the code I should modify to implement this feature.

Questions 2 :
Where in the codebase should I start to modify the extrinsic parameters for custom viewport rendering?
How should I handle the intrinsic parameters of the camera in this context? Should I adjust the focal length, principal point, etc., or can they be left unchanged?
Any advice or pointers would be greatly appreciated. Thanks again for your support and for creating such an incredible project!

Thanks again for your support and for creating such an incredible project!

**Comment by mkleijn** [comment]
Hi, I see a similar issue on a custom dataset (using no_prior, after 80k iterations), with blurry motions. 
It also looks like, at the last frames of the training section, there is something which looks like 'ghosting' from earlier frames. You can see the hair of all 3 the dancers move up a bit, as they were in some frames before. 

![dancers](https://github.com/user-attachments/assets/f4ff0bcf-b468-4b56-9a03-f2a8b29b6d16)

The point_cloud.ply and other files are available [here](https://drive.google.com/file/d/1e1WdOyT4Yv9cY67juMe3czGNXDRTe5Lw/view?usp=sharing)

is this a problem with the dataset, or is there a bug in the no_prior or training script or my choice of hyper-params? This is what I used as config:

{
    "resolution": 2,
    "model": "ours_lite",
    "scaling_lr": 0.005,
    "trbfslinit": 2.0,
    "preprocesspoints": 0,
    "test_iteration": 25000,
    "densify": 20,
    "desicnt": 12,
    "duration": 20,
    "rgbfunction": "None",
    "loader": "technicolor",
    "rdpip": "train_ours_lite"
  }

Any ideas what it could be causing these artificats?

**Comment by lizhan17** [comment]
> Hello, thank you for your amazing work on this project! I’ve been making good progress using the Techni full config for training on the Painter sequence. While the results are impressive for static backgrounds, I’m encountering an issue where dynamic objects (e.g., moving people) appear too blurry compared to the rest of the scene. I’ve experimented with the following, but I haven’t seen significant improvements:
> 
> Adjusting the training iterations (increased up to 200%). Tweaking various config parameters Applying different augmentation techniques, but the blurriness issue persists.
> 
> Question 1 : Could you please suggest any tips or potential adjustments I could try to resolve this issue? Are there specific config parameters or training strategies you recommend to handle dynamic objects more effectively?
> 
> ![00049](https://private-user-images.githubusercontent.com/101889605/385647454-cc62f84a-5379-446e-bda3-656a1fba1db8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1NTQyMjgsIm5iZiI6MTczMTU1MzkyOCwicGF0aCI6Ii8xMDE4ODk2MDUvMzg1NjQ3NDU0LWNjNjJmODRhLTUzNzktNDQ2ZS1iZGEzLTY1NmExZmJhMWRiOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExNFQwMzEyMDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jNmU5MGQ0ODgxY2I0MWZkYTI4NDkxNDcxOWRhYTAyMmVkNzE2NDQ4YWZlNTI2OGU3YTNkYjU5NTJmM2ZjMGY2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.oeYzoZc22YZn3yfqb1V3jAT1vutNagmSJelkYaKzX4w)
> 
> And I have one more question about custom viewport rendering with camera extrinsic parameters
> 
> I’d like to render a scene from a custom viewport by manually specifying the extrinsic parameters (X, Y, Z, Yaw, Pitch, Roll). I’m looking for guidance on which parts of the code I should modify to implement this feature.
> 
> Questions 2 : Where in the codebase should I start to modify the extrinsic parameters for custom viewport rendering? How should I handle the intrinsic parameters of the camera in this context? Should I adjust the focal length, principal point, etc., or can they be left unchanged? Any advice or pointers would be greatly appreciated. Thanks again for your support and for creating such an incredible project!
> 
> Thanks again for your support and for creating such an incredible project!

for rendering custom viewport. this is some code that we used to render moving view.

https://github.com/oppo-us-research/SpacetimeGaussians/blob/3213738943392b9a0a96ece1ab40cf993021aaa8/thirdparty/gaussian_splatting/scene/dataset_readers.py#L322

if you want to modify the extrinsic or intrinsic, directly modify the projecting matrix before it is fed to rendering call.

https://github.com/oppo-us-research/SpacetimeGaussians/blob/main/thirdparty/gaussian_splatting/renderer/__init__.py

**Comment by lizhan17** [comment]
> "duration": 20,

i think because this moving is large. poly motion function is not enough to fit the large motions. 
How many points in the cloud for a 50 frames ? i know dynamic3dgs-like perframe training may be good for large motions.

**Comment by mkleijn** [comment]
Hi @lizhan17, thanks for getting back so quickly. There are 233k points in the cloud. I will try training using [Dynamic3DGaussians](https://github.com/JonathonLuiten/Dynamic3DGaussians).

BTW as you quoted the "duration: 20" part of my config, would you suggest I try change that? And if so, would you suggest a higher or lower value?

**Comment by lizhan17** [comment]
we use 50 dataset we test. their motion is not large, but viewer is not 360 degree. 

if you use 20.   the initial temporal displacement is 1/20,  points are sparser than the default 1/50

**Comment by JunhyeongDoyle** [comment]
Additionally, I have a question regarding a slight resolution discrepancy between the input image and the rendered output. Specifically, the original Painter data has a resolution of 2048x1088, but the rendered output often appears at a slightly different resolution, like 2043x1085.

Do you know what might be causing this issue?

**Comment by lizhan17** [comment]
> Additionally, I have a question regarding a slight resolution discrepancy between the input image and the rendered output. Specifically, the original Painter data has a resolution of 2048x1088, but the rendered output often appears at a slightly different resolution, like 2043x1085.

The camera center is a float number with many demicals. I remember we use colmap's undistorting operation will have some loss of integer for width and height.

**Comment by lizhan17** [comment]
if we train longer frames, points at time 1 and points at time 300 will overlapped a lot in spatial, cause averaged opacity after training, if you also use frame1 and frame300 to traing them.   

we need a 1d gaussian window with a good shape for the motions (a shape/filter that can make point at time=1 disappear at point=300).  i think train with 50 frames can reduce this issue, multiple 50 frames with some trick to handle the cross sub-sequence difference.
